test/over_react/util/cast_ui_factory_test.dart

```diff
@@ -29,15 +29,19 @@ main() {
     });
 
     test('can use generics to set the type', () {
-      final testFactory = castUiFactory<UiFactory<BasicProps>>(_$Basic);
+      final testFactory = castUiFactory<BasicProps>(_$Basic);
 
       expect(testFactory, isA<UiFactory<BasicProps>>());
     });
 
     test('will leave the type a `UiFactory<UiProps` if no type is provided', () {
       final testFactory = castUiFactory(_$Basic);
 
-      expect(testFactory, isA<UiFactory<UiProps>>());
+      // This assignment should analyze properly without any static implicit_cast errors
+      UiFactory<UiProps> testAssignment = testFactory;
```

Comment:
Can we do this for the other tests this group? (except using <BasicProps> on the LHS)

In their current state, they're not exercising the typing